Modify WATSON_DIR on virtualenv activation

davidag opened the pull request:
I'm planning to contribute to watson, so I'm going through the repo and docs, trying to hack a bit in my local environment.

After reading the hack documentation and the issue #238 I think there is a potential risk of messing with your real watson data, which is not desired at all (although yes, I have backups :)

In this PR, I just make sure that `WATSON_DIR` will always be equal to `"$(watson_dir)/data"` when activating the virtualenv.

I can think of a couple of things against this PR: 
* The "data" directory name is only based on the comments of #238 but people could be using a different name. If this would be the case, it doesn't affect them because they're doing an export after activating the environment anyway.
* No documentation. I think the hack doc needs some work, but first PR #243 should be merged or rejected. After that, it would be nice to speak of the Makefile on that document, which would include a mention to this PR change. Also, adding Python3 support to the aforementioned Makefile could be interesting.

Any comments are welcomed!

jmaupetit commented on Makefile:
```diff
@@ -6,11 +6,13 @@ PIP ?= pip
 VENV = virtualenv
 VENV_ARGS = -p $(PYTHON)
 VENV_DIR = $(CURDIR)/.venv
+VENV_DATA_DIR = $(CURDIR)/data
```
Even if it will be loaded with the virtualenv, I think we can call it `WATSON_DIR` to be more explicit:
```suggestion
WATSON_DIR = $(CURDIR)/data
```

jmaupetit commented on Makefile:
```diff
@@ -6,11 +6,13 @@ PIP ?= pip
 
 $(VENV_DIR): requirements-dev.txt
 	$(VENV) $(VENV_ARGS) "$(VENV_DIR)"
+	echo "export WATSON_DIR=\"$(VENV_DATA_DIR)\"" >> "$(VENV_DIR)"/bin/activate
```
```suggestion
	echo "export WATSON_DIR=\"$(WATSON_DIR)\"" >> "$(VENV_DIR)"/bin/activate
```

davidag replied:
@jmaupetit what about `VENV_WATSON_DIR`? I find having two `WATSON_DIR` variables in two languages (shell & makefile) in the same expression a bit confusing. Besides, prefixing with `VENV` seems quite consistent and meaningful in that context. What do you think? If I didn't convince you I'll go with `WATSON_DIR` , no problem :)

jmaupetit replied:
I find this more consistent, but `VENV_WATSON_DIR` is ok to me (at least more relevant than `VENV_DATA_DIR` :wink:).

davidag commented:
I've modified the variable name and rebased onto master. I hope there's nothing missing or wrong, but if that would be the case, just let me know and I'll fix it. Thank you!
